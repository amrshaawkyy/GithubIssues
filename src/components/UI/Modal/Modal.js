import React, { Component } from 'react';
import './Modal.css';
import Aux from '../../../hoc/AUX1/Aux1';
import Backdrop from '../Backdrop/Backdrop';
import PropTypes from 'prop-types';
class Modal extends Component {
    shouldComponentUpdate(nextProps,nextState){
        return nextProps.show!==this.props.show || nextProps.children!==this.props.children;
    }
    render() {
        return (
            <Aux>
                <Backdrop show={this.props.show} clicked={this.props.modalClosed} />
                <div className="Modal"
                    style={{
                        transform: this.props.show ? 'translateY(0)' : 'translateY(-100vh)',
                        opacity: this.props.show ? '1' : '0'
                    }}>
                    {this.props.children}
                </div >
            </Aux>
        );
    }
}
Modal.propTypes = {
    show: PropTypes.bool,
    children: PropTypes.array,
    modalClosed: PropTypes.func
};
export default Modal;