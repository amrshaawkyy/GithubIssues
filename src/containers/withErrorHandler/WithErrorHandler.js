import React, { Component } from 'react';
import Modal from '../../components/UI/Modal/Modal';
import Aux from '../../hoc/AUX1/Aux1';
import PropTypes from 'prop-types';
class WithErrorHandler extends Component {
    state = {
        error: true
    }
    errorConfirmedHandler = () => {
        this.setState({ error: false });
    }
    render() {
        return (
            <Aux>
                <Modal show={this.state.error} modalClosed={this.errorConfirmedHandler} >
                    {this.state.error ? this.props.errormsg : null}
                </Modal>
            </Aux>
        );
    }
}
WithErrorHandler.propTypes = {
    errormsg: PropTypes.string
};
export default WithErrorHandler;
