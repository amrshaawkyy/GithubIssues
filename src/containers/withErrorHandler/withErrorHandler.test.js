import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import WithErrorHandler from './WithErrorHandler';
import Modal from '../../components/UI/Modal/Modal';

configure({adapter: new Adapter()});

describe('<WithErrorHandler />', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<WithErrorHandler />);
    });
    afterEach(() => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should render <Modal /> when error is true', () => {
        wrapper.setState({error: true});
        expect(wrapper.find(Modal).length).toBe(1);
    });
});


