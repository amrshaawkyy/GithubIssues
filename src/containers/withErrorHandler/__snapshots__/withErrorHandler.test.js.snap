// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WithErrorHandler /> should render <Modal /> when error is true 1`] = `ShallowWrapper {}`;
